ansible_collections/arista/avd/plugins/filter/convert_dicts.py

```diff
@@ -58,8 +58,21 @@ def convert_dicts(dictionary, primary_key="name"):
                 # Catch cornercase where dictionary has no value because of old data models
                 output.append({primary_key: key})
             elif not isinstance(dictionary[key], dict):
-                # Not a nested dictionary, return the original
-                return dictionary
+                if isinstance(dictionary[key], list):
+                    # Not a nested dictionary but a list, add secondary key for inner list
+                    if secondary_key is not None:
+                        item = {}
+                        secondary_element = []
+                        item.update({primary_key: key})
+                        for element in dictionary[key]:
+                            secondary_element.append(element)
+                        item.update({secondary_key: secondary_element})
+                        output.append(item)
+                    else:
+                        raise Exception("Argument required : secondary_key")
```

Comment:
lets not error out, but just return the original dictionary like before.

Reply:
returning the original dict below outside this if condition. This is if secondary_key is not passed in argument in case of list. shall we remove the error here?

Reply:
Let's add a default `items` value to the secondary key, like we have for primary key. Then this case will not happen.